test/parallel_api/ranges/std_ranges_find_first_of.pass.cpp

```diff
@@ -0,0 +1,34 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "std_ranges_test.h"
+
+std::int32_t
+main()
+{
+#if _ENABLE_STD_RANGES_TESTING
+    using namespace test_std_ranges;
+    namespace dpl_ranges = oneapi::dpl::ranges;
+
+    auto find_first_of_checker = TEST_PREPARE_CALLABLE(std::ranges::find_first_of);
+
+    test_range_algo<0, int, data_in_in>{big_sz}(dpl_ranges::find_first_of, find_first_of_checker, binary_pred);
+    test_range_algo<1, int, data_in_in>{}(dpl_ranges::find_first_of, find_first_of_checker, binary_pred, proj, proj);
+    test_range_algo<2, P2, data_in_in>{}(dpl_ranges::find_first_of, find_first_of_checker, binary_pred, &P2::x, &P2::x);
+    test_range_algo<3, P2, data_in_in>{}(dpl_ranges::find_first_of, find_first_of_checker, binary_pred, &P2::proj, &P2::proj);
```

Comment:
Can you check a case with different-sized ranges?
Will it work as expected if the second sequence is empty?

Reply:
It is impossible here (in this file).
This test file relies on standard test data sized defined in common test header, where the sizes are various following way:

```
        process_data_in_out(max_n, r_size/2, r_size, exec, algo, checker, args...);
        process_data_in_out(max_n, r_size, r_size/2, std::forward<Policy>(exec), algo, checker, args...);
```

Reply:
Yeah... good catch

Reply:
1) added `process_data_in_in`.
2) The errors in `find_end `impl have been fixed.